gk/main.c

```diff
@@ -850,6 +850,83 @@ add_ggu_policy(struct ggu_policy *policy,
 	}
 }
 
+/*
+ * When a network administrator changes a policy in response to an attack,
+ * she may need to flush all policy decisions cached at the Gatekeeper servers
+ * associated to a given destination prefix in order to quickly re-establish the
+ * network. Function flush_ggu_policy() supports these flush requests.
+ */
+static void
+flush_ggu_policy(struct ggu_policy *policy, struct gk_instance *instance)
+{
+	uint8_t prefix_len = policy->params.flushed.prefix_len;
+	uint32_t next = 0;
+	int32_t index;
+	const struct ip_flow *key;
+	void *data;
+	struct in_addr ip4_mask;
+	struct in6_addr ip6_mask;
+
+	memset(&ip4_mask, 0, sizeof(ip4_mask));
+	memset(&ip6_mask, 0, sizeof(ip6_mask));
+
+	if (policy->flow.proto == ETHER_TYPE_IPv4) {
+		ip4_mask.s_addr =
+			rte_cpu_to_be_32(~0ULL << (32 - prefix_len));
+	} else if (likely(policy->flow.proto == ETHER_TYPE_IPv6)) {
```

Comment:
Don't we already have a function for generating a mask from a prefix length? I think I've reviewed a code like this somewhere else. Let's avoid the duplicate code. If needed, create the function and use it where needed.